registration_2019/dayof.py

```diff
@@ -6,27 +6,46 @@
 from .core import api, db, app
 from .helper import *
 from .authentication import auth
-from .registration import Signup, SignIn
+from .registration import Signup
+from .mentor import Mentor
+from .guest import Guest
+from .dayof_model import SignIn
 
-## Helpers
-
-def validate_badge_data(badge_data):
-    # TODO validate badge_data based on whatever method we decide
-    return True
+# TODO write actual regex
+badge_data = re_matches(".*", "badge data")
 
 def sign_in(user_id, badge_data):
-    if not validate_badge_data(badge_data):
-        return {"message": "Invalid badge data"}, 400
     signup = Signup.query.filter_by(user_id=user_id, outdated=False).scalar()
-    if signup and signup.sign_in:
-        return {"message": "User already signed in"}, 400
-    sign_in = SignIn(badge_data=badge_data)
-    db.session.add(sign_in)
-    # Could use the modify method, but interaction would be non-trivial given that its designed for forward facing API
-    signup.sign_in = sign_in
-    db.session.commit()
-
-    return {"status": "ok"}
+    if signup:
+        if signup.sign_in:
```

Comment:
If the SignIn has `signed_out` true, this will return that they have already signed in?

Reply:
I interpreted sign in and sign out as a one time thing, if we wanted to add tracking whether they are currently at the event then I would need to add more code.